argo/providers/stub/provider.go

```diff
@@ -71,31 +73,31 @@ func (s StubProvider) DeletePipeline(_ context.Context, providerConfig StubProvi
 }
 
 func (s StubProvider) CreateRun(_ context.Context, providerConfig StubProviderConfig, resourceDefinition base.RunDefinition) (string, error) {
-	return verifyCreateCall(providerConfig, ResourceDefinition{resourceDefinition.Name.Name, resourceDefinition.Version})
+	return verifyCreateCall(providerConfig, ResourceDefinition{resourceDefinition.Name, resourceDefinition.Version})
 }
 
 func (s StubProvider) DeleteRun(_ context.Context, providerConfig StubProviderConfig, id string) error {
 	return verifyDeleteCall(providerConfig, id)
 }
 
 func (s StubProvider) CreateRunSchedule(_ context.Context, providerConfig StubProviderConfig, resourceDefinition base.RunScheduleDefinition) (string, error) {
-	return verifyCreateCall(providerConfig, ResourceDefinition{resourceDefinition.Name, resourceDefinition.Version})
+	return verifyCreateCall(providerConfig, ResourceDefinition{common.NamespacedName{Name: resourceDefinition.Name, Namespace: providerConfig.Namespace}, resourceDefinition.Version})
```

Comment:
Initially it was because of the experiment and the stub provider and not having the namespace available in the resource but I think the issue is probably that we set the ExpectedInput incorrectly in the integration test for experiment to have a namespace. No work has been done to namespace experiments. Will need to look into it

Reply:
Fixed in [8f7d8ac](https://github.com/sky-uk/kfp-operator/pull/326/commits/8f7d8ac652bebd4e69b9f9f0f07ca0abf89b1c42)